homeassistant/components/tplink/light.py

```diff
@@ -84,6 +84,16 @@ class LightState(NamedTuple):
     hs: Tuple[int, int]
     emeter_params: dict
 
+    def to_param(self):
+        """Return a version that we can send to the bulb."""
+        return {
+            "on_off": 1 if self.state else 0,
+            "brightness": brightness_to_percentage(self.brightness),
+            "color_temp": mired_to_kelvin(self.color_temp) if self.color_temp else None,
+            "hue": self.hs[0],
```

Comment:
```suggestion
            "hue": int(self.hs[0]),
```

Reply:
The hs tuple is defined as Tuple[int, int], so there should be no need for casting..

Reply:
```2020-04-06 22:12:44 DEBUG (SyncWorker_11) [custom_components.tplink.light] sending new state: {'hue': 75.0, 'saturation': 85.71428571428571, 'color_temp': 0}
```

Without the casting we get this

Reply:
In `async_turn_on` we take the parameters directly from kwargs, we should cast them there. Making a patch.